if mapped initial commit

scottaohara opened the pull request:
* adds bullet to element role mappings
* adds TODO add if mapped for ia2/uia
* adds if mapped conditional for ATK mappings
* minor cleanups

Work to be done:
- [ ] similar wording and mapping expectations for IA2 / UIA (see todo comments)
- [x] revise wording for `div` & any other 'block' elements that should be updated to match

Closes #151 

The following tasks have been completed:
 * [x] Confirmed there are no ReSpec/BikeShed errors or warnings.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/pull/226.html" title="Last updated on Jul 28, 2019, 3:29 PM UTC (a0ae511)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/226/f7c394a...a0ae511.html" title="Last updated on Jul 28, 2019, 3:29 PM UTC (a0ae511)">Diff</a>

scottaohara committed:
if mapped initial commit

* adds bullet to element role mappings
* adds TODO add if mapped for ia2/uia
* adds if mapped conditional for ATK mappings
* minor cleanups

scottaohara commented:
Initial draft, please take a look if you can @joanmarie @carmacleod @stevefaulkner to make sure i'm on the right track with this.

More work needs to happen here, but I wanted to at least get the verbage in for ATK from @joanmarie.  Needs to add ia2/uia for other span-like elements and need to update the wording for elements like `div` that are "not mapped" / "if mapped"

scottaohara committed:
update url of link ref

carmacleod commented:
Hi, @scottaohara !
There's something funny with the Preview and Diff links - they don't include the table(s)... and it's a little bit rough to review the changes in the code because one needs to expand previous lines of code to figure out which element's table row each change is in... :)
Is it easy to fix the missing table in the Preview | Diff links?
Or do you have any other preview/diff links?

That said, I think it's on the right track from what I could see.

scottaohara commented:
@marcoscaceres might the preview/diff be an issue with the custom script that converts the giant table into individual element disclosure widgets?

just wonder if there might be a way to fix that / at the very least look into just exposing the full table in this view.

scottaohara committed:
cleanup

scottaohara committed:
cleanup formatting

scottaohara committed:
If mapped

* move “if mapped” prior to the role information.
* `b` AX update to “if mapped”.  Otherwise only the AXStaticText subtree is exposed.
* update ATK and AX mappings for `data` if mapped.
* update introduction of `div` mappings to “if mapped”
* `strong` AX update to “if mapped”. Otherwise only the AXStaticText subtree is exposed.

scottaohara committed:
update AX mappings

* update `i` to “if mapped”
* update `s` to “if mapped”
* update `span` to “if mapped”

note: reverified that `var`, `kbd`, `samp`, `code` all map to `AXGroup` without assistance from an attribute like `aria-label` or `tabindex` to promote them as an accessible object.

scottaohara committed:
u cleanup

scottaohara committed:
add AX mapping for <u>

scottaohara committed:
edit wording for span if mapped atk

scottaohara committed:
AX small update

note that only if mapped

scottaohara committed:
add bdi AX mappings

scottaohara committed:
update bdo AX mapping

note that AXGroup only if mapped

carmacleod commented on index.html:
```diff
@@ -3184,10 +3276,10 @@ <h3>HTML Element Role Mappings</h3>
                 <td class="ia2">
-                  <!-- add iff mapped -->
                   <div class="general">No accessible object.</div>
+                  <!-- add if mapped -->
```
might want to add TODO so you find this later :)

carmacleod commented on index.html:
```diff
@@ -3535,23 +3724,55 @@ <h3>HTML Element Role Mappings</h3>
+              <td class="aria">No corresponding role</td>
+              <td class="ia2">
+                <div class="general">
+                  If mapped:
```
I don't think the MSAA or UIA columns for `u` need the "If mapped: ... Otherwise".
I think they should be similar wording/formatting to the MSAA and UIA columns for `s`?

carmacleod commented:
Can I help with this? I can open a new issue for my last review comment (about u element) if that's holding things up.

scottaohara commented:
Hi @carmacleod 

No your comment wasn't holding things up.  That was just a part of the spec I hadn't finished updating yet.

It's more that as time has passed, I've become unsure that this is the direction this should be taking. Especially with how things have progressed with ARIA, and many of these elements should have a mapping of `role=generic`.

carmacleod commented:
Ah, of course. Thanks, @scottaohara 
I was asking because the issue regarding adding hgroup and slot to HTML AAM says https://github.com/w3c/html-aam/issues/189#issuecomment-517949442.

scottaohara commented:
closing (but not deleting) this pull request as mappings to `generic` will make this unnecessary.
